crates/turborepo-filewatch/src/package_watcher.rs

```diff
@@ -468,69 +468,72 @@ impl<T: PackageDiscovery + Send + Sync + 'static> Subscriber<T> {
             .filter_map(|p| p.as_os_str().to_str())
         {
             let path_file = AbsoluteSystemPathBuf::new(path).expect("watched paths are absolute");
-
-            // the path to the workspace this file is in is the parent
-            let path_workspace = path_file
-                .parent()
-                .expect("watched paths will not be at the root")
-                .to_owned();
-
-            let is_workspace = match state
-                .filter
-                .target_is_workspace(&self.repo_root, &path_workspace)
-            {
-                Ok(is_workspace) => is_workspace,
-                Err(e) => {
-                    // this will only error if `repo_root` is not an anchor of `path_workspace`.
-                    // if we hit this case, we can safely ignore it
-                    tracing::debug!("yielded path not in workspace: {:?}", e);
+            let path_workspace = if path_file.file_name() == Some("package.json") {
```

Comment:
In no way blocking, but `path_workspace` can be `&AbsoluteSystemPath` which will avoid an allocation in the "package.json is no longer on disk" path. (This does require deriving `Hash` on `AbsoluteSystemPath`, but that's something that should have probably already been derived)

Reply:
Done.